service_configuration_lib/spark_config.py

```diff
@@ -163,6 +169,23 @@ def _get_mesos_docker_volumes_conf(
     return {'spark.mesos.executor.docker.volumes': volume_str}
 
 
+def _get_k8s_docker_volumes_conf(
+    volumes: Optional[List[Mapping[str, str]]] = None,
+):
+    env = {}
+    k8s_volumes = volumes or []
+    k8s_volumes.append(DEFAULT_SPARK_K8S_SECRET_VOLUME)
```

Comment:
I just realized this is confusing. 
For consistency, let's either remove DEFAULT_SPARK_K8S_SECRET_VOLUME to change to format k8s_volumes.append({'containerPath': '/etc/group', 'hostPath': '/etc/group', 'mode': 'RO'})
or the other way around?

Or is there a reason for two different patterns for appending?